csharp/test/Microsoft.ML.OnnxRuntime.Tests.Common/InferenceTest.cs

```diff
@@ -20,90 +20,94 @@
+        opt.DisablePerSessionThreads();
 
-                // try setting options
-                opt.ExecutionMode = ExecutionMode.ORT_PARALLEL;
-                Assert.Equal(ExecutionMode.ORT_PARALLEL, opt.ExecutionMode);
+        // try setting options
+        opt.ExecutionMode = ExecutionMode.ORT_PARALLEL;
+        Assert.Equal(ExecutionMode.ORT_PARALLEL, opt.ExecutionMode);
 
-                opt.EnableMemoryPattern = false;
-                Assert.False(opt.EnableMemoryPattern);
+        opt.EnableMemoryPattern = false;
+        Assert.False(opt.EnableMemoryPattern);
 
-                opt.EnableProfiling = true;
-                Assert.True(opt.EnableProfiling);
-                Assert.Equal("onnxruntime_profile_", opt.ProfileOutputPathPrefix);
+        opt.EnableProfiling = true;
+        Assert.True(opt.EnableProfiling);
+        Assert.Equal("onnxruntime_profile_", opt.ProfileOutputPathPrefix);
 
-                opt.ProfileOutputPathPrefix = "Ort_P_";
-                Assert.Equal("Ort_P_", opt.ProfileOutputPathPrefix);
+        opt.ProfileOutputPathPrefix = "Ort_P_";
+        Assert.Equal("Ort_P_", opt.ProfileOutputPathPrefix);
 
-                opt.EnableCpuMemArena = false;
-                Assert.False(opt.EnableCpuMemArena);
+        opt.EnableCpuMemArena = false;
+        Assert.False(opt.EnableCpuMemArena);
 
-                opt.LogId = "MyLogId";
-                Assert.Equal("MyLogId", opt.LogId);
+        opt.LogId = "MyLogId";
+        Assert.Equal("MyLogId", opt.LogId);
 
-                opt.LogVerbosityLevel = 1;
-                Assert.Equal(1, opt.LogVerbosityLevel);
+        opt.LogVerbosityLevel = 1;
+        Assert.Equal(1, opt.LogVerbosityLevel);
 
-                opt.LogSeverityLevel = OrtLoggingLevel.ORT_LOGGING_LEVEL_WARNING;
-                Assert.Equal(OrtLoggingLevel.ORT_LOGGING_LEVEL_WARNING, opt.LogSeverityLevel);
+        opt.LogSeverityLevel = OrtLoggingLevel.ORT_LOGGING_LEVEL_WARNING;
+        Assert.Equal(OrtLoggingLevel.ORT_LOGGING_LEVEL_WARNING, opt.LogSeverityLevel);
 
-                opt.IntraOpNumThreads = 4;
-                Assert.Equal(4, opt.IntraOpNumThreads);
+        opt.IntraOpNumThreads = 4;
+        Assert.Equal(4, opt.IntraOpNumThreads);
 
-                opt.InterOpNumThreads = 4;
-                Assert.Equal(4, opt.InterOpNumThreads);
+        opt.InterOpNumThreads = 4;
+        Assert.Equal(4, opt.InterOpNumThreads);
 
-                opt.GraphOptimizationLevel = GraphOptimizationLevel.ORT_ENABLE_EXTENDED;
-                Assert.Equal(GraphOptimizationLevel.ORT_ENABLE_EXTENDED, opt.GraphOptimizationLevel);
+        opt.GraphOptimizationLevel = GraphOptimizationLevel.ORT_ENABLE_EXTENDED;
+        Assert.Equal(GraphOptimizationLevel.ORT_ENABLE_EXTENDED, opt.GraphOptimizationLevel);
 
-                Assert.Throws<OnnxRuntimeException>(() => { opt.GraphOptimizationLevel = (GraphOptimizationLevel)10; });
+        AssertUtils.IfThrowsCheckException<OnnxRuntimeException>(
```

Comment:
if I understand correctly, now the test will not fail if an exception is not thrown by the provided action.

why are these statements expected to possibly not throw an exception now?